pkg/network/node/networkpolicy.go

```diff
@@ -314,25 +346,39 @@ func (np *networkPolicyPlugin) SyncVNIDRules() {
 // Yes, if a selector matches against multiple labels then the order they appear in
 // cacheKey here is non-deterministic, but that just means that, eg, we might compute the
 // results twice rather than just once, and twice is still better than 10,000 times.
-func (np *networkPolicyPlugin) selectNamespacesInternal(sel labels.Selector) map[string]uint32 {
-	cacheKey := sel.String()
-	namespaces, wasCached := np.nsMatchCache[cacheKey]
-	if !wasCached {
-		namespaces = make(map[string]uint32)
-		for vnid, ns := range np.namespaces {
-			if kns, exists := np.kNamespaces[ns.name]; exists {
-				if sel.Matches(labels.Set(kns.Labels)) {
-					namespaces[ns.name] = vnid
-				}
+func (np *networkPolicyPlugin) selectNamespacesInternal(selector labels.Selector) map[string]uint32 {
+	cacheKey := selector.String()
+	match := np.nsMatchCache[cacheKey]
+	if match == nil {
+		match = &npCacheEntry{selector: selector, matches: make(map[string]uint32)}
+		for vnid, npns := range np.namespaces {
+			if npns.gotNamespace && selector.Matches(labels.Set(npns.labels)) {
```

Comment:
does this need to be `gotNamespace && gotNetNamespace` ?

Reply:
no; we're iterating over `np.namespaces`, which is indexed by vnid and so necessarily only contains the namespaces for which `gotNetNamespace` is `true`.

Reply:
oh duh. good point.